src/main/java/graphql/schema/validation/DeprecatedInputObjectAndArgumentsAreValid.java

```diff
@@ -0,0 +1,63 @@
+package graphql.schema.validation;
+
+import graphql.Internal;
+import graphql.schema.GraphQLAppliedDirective;
+import graphql.schema.GraphQLArgument;
+import graphql.schema.GraphQLDirective;
+import graphql.schema.GraphQLFieldDefinition;
+import graphql.schema.GraphQLInputObjectField;
+import graphql.schema.GraphQLInputObjectType;
+import graphql.schema.GraphQLSchemaElement;
+import graphql.schema.GraphQLTypeUtil;
+import graphql.schema.GraphQLTypeVisitorStub;
+import graphql.util.TraversalControl;
+import graphql.util.TraverserContext;
+
+import static java.lang.String.format;
+
+/*
+  From the spec:
+  The @deprecated directive must not appear on required (non-null without a default) arguments
+  or input object field definitions.
+ */
+@Internal
+public class DeprecatedInputObjectAndArgumentsAreValid extends GraphQLTypeVisitorStub {
+
+    @Override
+    public TraversalControl visitGraphQLInputObjectField(GraphQLInputObjectField inputObjectField, TraverserContext<GraphQLSchemaElement> context) {
+        // There can only be at most one @deprecated, because it is not a repeatable directive
+        GraphQLAppliedDirective deprecatedDirective = inputObjectField.getAppliedDirective("deprecated");
```

Comment:
Use `Directives.Deprecated.getName()` perhaps to stop string spread

Reply:
Yes that's a better idea, I'll add that. Don't want a random string hanging out here

Reply:
I fixed both strings in the commit after your review